ffi/src/lib.rs

```diff
@@ -153,10 +163,10 @@ mod private {
         /// The slice must have been originally created `From<Vec<bool>>`, and must not have been
         /// already been consumed by a previous call to this method.
         pub unsafe fn as_ref(&self) -> &[bool] {
-            if self.ptr.is_null() {
+            if self.len == 0 {
```

Comment:
From what I understand, the pointer is always valid. The len just says how much memory is safe to reference (so a dangling pointer _is_ valid for len zero, because no "safe to touch" memory is needed or implied).

Reply:
Yup, we can remove the if statement done